src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,46 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+
+    protected String by;
+    protected String final_format;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    protected void date_formatter() {
```

Comment:
Can consider naming formatDate, since Java coding standards states that names representing methods must be verbs and written in camelCase